packages/dds/tree/src/feature-libraries/typed-schema/typedTreeSchema.ts

```diff
@@ -192,6 +194,11 @@ export class FlexObjectNodeSchema<
 	) {
 		const fields = mapIterable(objectNodeFields, ([k, v]) => [k, v.stored] as const);
 		super(builder, name, info, new ObjectNodeStoredSchema(new Map(fields)));
+		objectNodeFields.forEach((flexFieldSchema, fieldKey) => {
```

Comment:
`identifierFieldKeys` should be a `readonly FieldKey[]` (in addition to the field itself being readonly). You can do this by making `identifierFieldKeys` private and adding a public getter with the narrower type. Or, if you want to have a little bit more fun and remove some lines, you could add a `filterIterable` function alongside our existing `mapIterable` function in our top-level utilities. Then here you'd just do:

```ts
this.identifierFieldKeys = Array.from(filterIterable(objectNodeFields.entries(), ([k, f]) => f.kind.identifier === identifierFieldKindIdentifier), ([k]) => k);
```

and no getter needed.

Reply:
Added `filterIterable` function in `utils.ts` and simplified code

Reply:
Nice, way to go the extra mile!! Don't forget to make `identifierFieldKeys` a `readonly Field[]` (that's one of the benefits of initializing it in this "functional" style :))

Reply:
Updated `identifierFieldKeys` to be `readonly Field[]`